src/platform/modulesmanager/tests/ModuleManagerTests/src/MpiTests.cpp

```diff
@@ -107,4 +107,42 @@ namespace Tests
         std::string payloadStr(payload, szPayload);
         ASSERT_TRUE(Tests::JSON_EQ(expectedPayload, payloadStr.c_str()));
     }
+
+    TEST_F(MpiTests, MpiSetDesired)
+    {
+        std::string clientName = "MpiTests.MpiSetDesired";
+        char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
+                }
+            ])"""";
```

Comment:
Insufficient, please see other comments

Reply:
Updated tests to clarify the use of objects vs settings.